FreeRTOS-Plus/Demo/coreHTTP_Windows_Simulator/HTTP_S3_Download/DemoTasks/S3DownloadHTTPExample.c

```diff
@@ -750,6 +753,24 @@ static BaseType_t prvDownloadS3ObjectFile( const TransportInterface_t * pxTransp
 
         if( xHTTPStatus == HTTPSuccess )
         {
+            /* Check for the "Connection: close" header in case the connection
+             * needs to be re-established. */
+            if( xResponse.respFlags == HTTP_RESPONSE_CONNECTION_CLOSE_FLAG )
```

Comment:
Why are we assuming this is happening because of some request limit?  What if the bucket/file doesn't exist, there is a policy issue, etc?  I think you'd see similar behavior.  This seems like __a bad idea__ – so why are we doing it?

Same comment at 1134.

Reply:
This is resolved with checking for the status code before doing anything else.